### PR TITLE
update the external links that contains master to release-4.0

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Update the external GitHub links that contain `/master/` and `/release-3.0/` to `/release-4.0/`.

### Which TiDB version(s) do your changes apply to? (Required)

- [ ] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs-cn/pull/6054

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: explain-overview.md [hunk 1]
```diff
@@ -343,7 +343,7 @@ mysql> EXPLAIN SELECT /*+ INL_MERGE_JOIN(t1, t2) */ * FROM t1, t2 WHERE t1.id =
 
 ## 优化实例
 
-使用 [bikeshare example database](https://github.com/pingcap/docs/blob/release-4.0/import-example-data.md):
+使用 [bikeshare example database](https://docs.pingcap.com/tidb/v4.0/import-example-data):
```
Comment (yikeke) [inline explain-overview.md:346]: 这个保持原样吧

**Review by yikeke — APPROVED** [review]

File: explain-overview.md (hunk 1 above)
Comment (TomShawn) [reply explain-overview.md:346]:
```suggestion
使用 [bikeshare example database](https://github.com/pingcap/docs/blob/release-4.0/import-example-data.md):
```

**Pull request merged**